py-shared/src/quilt_shared/aws.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+import typing as T
+
+import pydantic
+
+from .types import NonEmptyStr
+
+if T.TYPE_CHECKING:
+    import boto3
+
+
+class AWSCredentials(pydantic.BaseModel):
+    key: NonEmptyStr
+    secret: NonEmptyStr
+    token: NonEmptyStr
+
+    @property
+    def boto_args(self):
+        return dict(
+            aws_access_key_id=self.key,
+            aws_secret_access_key=self.secret,
+            aws_session_token=self.token,
+        )
+
+    @classmethod
+    def from_boto_session(cls, session: boto3.Session):
+        credentials = session.get_credentials()
+        assert credentials
```

Comment:
maybe raise `ValueError`? it's really can be `None`

Reply:
ok